File: infra/src/main/java/mixter/infra/InMemoryEventStore.java
package mixter.infra;

import mixter.domain.AggregateId;
import mixter.domain.Event;

import java.util.ArrayList;
import java.util.List;

public class InMemoryEventStore implements EventStore {
    List<Event> events = new ArrayList<>();

    @Override
    public List<Event> getEventsOfAggregate(AggregateId aggregateId) {
        return events;
    }

    @Override
    public void store(Event event) {
        events.add(event);
    }
}

File: src/main/scala/infra/InMemoryEventStore.scala
package infra
import domain.{AggregateId, Event}

private[infra] class InMemoryEventStore extends EventStore {
  var events = Seq.empty[Event]

  override def store(event: Event): Unit =
    events = events :+ event

  override def eventsOfAggregate(aggregateId: AggregateId): Seq[Event] =
    events

}
